source/common/modules/markdown-editor/plugins/visual-indent.ts

```diff
@@ -15,167 +15,121 @@
 
-    const columnLineTextStart = spaces + tabs + listMarker
-    // const visualLineTextStart = spaces + tabs * tabSize + listMarker
+      const columnLineTextStart = leadingWhitespace + formatCharOffset
 
-    if (columnLineTextStart === 0) {
-      continue // Neither whitespace nor list elements on that line.
-    }
+      if (columnLineTextStart === 0) {
+        continue // Neither whitespace nor list elements on that line.
+      }
 
-    // Now that we know we need to indent this line, schedule a measurement so
-    // that in the next round of this code running we have an accurate
-    // indentation even for non-monospaced text. What we want to measure is
-    // exclusively what we are basing our indentation on. If we use the entire
-    // line as a key, we cause (a) duplicated measurements (`* One` and `* Two`
-    // require the same indentation) and (b) cause cache misses which results in
-    // jumpy behavior when the user adds novel characters to the line.
-    const measurementKey = line.text.slice(0, columnLineTextStart).replace('\t', ' '.repeat(tabSize))
-
-    view.requestMeasure({
-      read (view) {
-        const base = view.contentDOM.getBoundingClientRect().left
-        const after = view.coordsAtPos(line.from + columnLineTextStart)?.left ?? 0
-        if (after === 0) {
-          return // Could not retrieve coordinates
-        }
-        // Note that this continuously updates our measurements after any layout
-        // changes
-        measurements?.set(measurementKey, after - base)
-      },
-      key: measurementKey
-    })
-
-    const indent = measurements?.get(measurementKey)
-    if (indent !== undefined) {
-      // NOTE: Each `.cm-line` has a padding of `0 2px 0 6px` as per CodeMirror's
-      // base styles from somewhere in the library. We need to account for that
-      // not to induce any problems.
-      const basePadding = 6
-      const deco = Decoration.line({ attributes: { style: `text-indent: -${indent-basePadding}px; padding-left: ${indent}px;` } })
-      ranges.push(deco.range(line.from))
+      // Now that we know we need to indent this line, schedule a measurement so
+      // that in the next round of this code running we have an accurate
+      // indentation even for non-monospaced text. What we want to measure is
+      // exclusively what we are basing our indentation on. If we use the entire
+      // line as a key, we cause (a) duplicated measurements (`* One` and `* Two`
+      // require the same indentation) and (b) cause cache misses which results in
+      // jumpy behavior when the user adds novel characters to the line.
+      const measurementKey = line.text.slice(0, columnLineTextStart).replace('\t', ' '.repeat(view.state.tabSize))
+
+      view.requestMeasure({
+        read (view) {
+          const pos = line.from + columnLineTextStart
+
+          const folded = foldedRanges(view.state).iter(pos)
```

Comment:
Please add a comment describing that you're here searching for folded regions to not indent those accidentally